src/table/tableview/examples/table-view-basic.js

```diff
@@ -0,0 +1,185 @@
+  * See {@link patternfly.toolbars.componenet:pfToolbar pfToolbar} for use in Toolbar View Switcher
+  *
+  * @param {object} config Optional configuration object
+  * <ul style='list-style-type: none'>
+  *   <li>.selectionMatchProp     - (string) Property of the items to use for determining matching, default is 'uuid'
+  *   <li>.onCheckBoxChange       - ( function(item) ) Called to notify when a checkbox selection changes, default is none
+  * </ul>
+  * @param {object} dtOptions Optional angular-datatables DTOptionsBuilder configuration object.  See {@link http://l-lin.github.io/angular-datatables/archives/#/api angular-datatables: DTOptionsBuilder}
+  * @param {array} items Array of items to display in the table view.
+  * @param {array} columns Array of table column information to display in the table's header row
+  * <ul style='list-style-type: none'>
+  *   <li>.header     - (string) Text label for a column header
+  *   <li>.itemField    - (string) Item field to associate with a particular column.
+  * </ul>
+  * @param {array} actionButtons List of action buttons in each row
+  *   <ul style='list-style-type: none'>
+  *     <li>.name - (String) The name of the action, displayed on the button
+  *     <li>.title - (String) Optional title, used for the tooltip
+  *     <li>.actionFn - (function(action)) Function to invoke when the action selected
+  *   </ul>
+  * @param {array} menuActions List of actions for dropdown menu in each row
+  *   <ul style='list-style-type: none'>
+  *     <li>.name - (String) The name of the action, displayed on the button
+  *     <li>.title - (String) Optional title, used for the tooltip
+  *     <li>.actionFn - (function(action)) Function to invoke when the action selected
+  *   </ul>
+  * @example
+ <example module="patternfly.table">
+ <file name="index.html">
+ <div ng-controller="TableCtrl" class="row example-container">
+   <div class="col-md-12">
+     <pf-table-view id="exampleTableView"
+          config="config"
+          dt-options="dtOptions"
+          colummns="colummns"
+          items="items"
+          action-buttons="actionButtons"
+          menu-actions="menuActions">
+     </pf-table-view>
+     <div class="col-md-12" style="padding-top: 12px;">
+       <label style="font-weight:normal;vertical-align:center;">Events: </label>
+     </div>
+     <div class="col-md-12">
+       <textarea rows="10" class="col-md-12">{{eventText}}</textarea>
+     </div>
+   </div>
+ </div>
+ </file>
+
+ <file name="script.js">
+ angular.module('patternfly.table').controller('TableCtrl', ['$scope',
+ function ($scope) {
+        $scope.dtOptions = {
+          order: [[2, "asc"]],
+        };
+
+        $scope.colummns = [
+          { header: "Name", itemField: "name" },
+          { header: "Address", itemField: "address"},
+          { header: "City", itemField: "city" },
+          { header: "State", itemField: "state"}
+        ];
+
+        $scope.items = [
```

Comment:
Datatables.net supports different types of data sources. For example:

- HTML (DOM) sourced data
- Ajax sourced data
- Javascript sourced data
- Server-side processing

This example appears to be using the Javascript approach. This is fine, but means the table must be provided with all data upfront (i.e., for pagination to function properly).

Does this implementation also support HTML and/or Ajax sourced data?